app/src/main/java/de/danoeh/antennapod/menuhandler/FeedItemMenuHandler.java

```diff
@@ -31,108 +33,133 @@
+        setItemVisibility(menu, R.id.share_file, fileDownloaded);
 
-        mi.setItemVisibility(R.id.remove_new_flag_item, selectedItem.isNew());
+        setItemVisibility(menu, R.id.remove_new_flag_item, selectedItem.isNew());
         if (selectedItem.isPlayed()) {
-            mi.setItemVisibility(R.id.mark_read_item, false);
+            setItemVisibility(menu, R.id.mark_read_item, false);
         } else {
-            mi.setItemVisibility(R.id.mark_unread_item, false);
+            setItemVisibility(menu, R.id.mark_unread_item, false);
         }
 
         if(selectedItem.getMedia() == null || selectedItem.getMedia().getPosition() == 0) {
-            mi.setItemVisibility(R.id.reset_position, false);
+            setItemVisibility(menu, R.id.reset_position, false);
         }
 
         if(!UserPreferences.isEnableAutodownload() || fileDownloaded) {
-            mi.setItemVisibility(R.id.activate_auto_download, false);
-            mi.setItemVisibility(R.id.deactivate_auto_download, false);
+            setItemVisibility(menu, R.id.activate_auto_download, false);
+            setItemVisibility(menu, R.id.deactivate_auto_download, false);
         } else if(selectedItem.getAutoDownload()) {
-            mi.setItemVisibility(R.id.activate_auto_download, false);
+            setItemVisibility(menu, R.id.activate_auto_download, false);
         } else {
-            mi.setItemVisibility(R.id.deactivate_auto_download, false);
+            setItemVisibility(menu, R.id.deactivate_auto_download, false);
+        }
+
+        // Display proper strings when item has no media
+        if (!hasMedia && !selectedItem.isPlayed()) {
+            changeStringNoMedia(menu, R.id.mark_read_item, R.string.mark_read_no_media_label);
+        }
+
+        if (!hasMedia && selectedItem.isPlayed()) {
+            changeStringNoMedia(menu, R.id.mark_unread_item, R.string.mark_unread_label_no_media);
         }
 
         boolean isFavorite = selectedItem.isTagged(FeedItem.TAG_FAVORITE);
-        mi.setItemVisibility(R.id.add_to_favorites_item, !isFavorite);
-        mi.setItemVisibility(R.id.remove_from_favorites_item, isFavorite);
+        setItemVisibility(menu, R.id.add_to_favorites_item, !isFavorite);
+        setItemVisibility(menu, R.id.remove_from_favorites_item, isFavorite);
 
-        mi.setItemVisibility(R.id.remove_item, fileDownloaded);
+        setItemVisibility(menu, R.id.remove_item, fileDownloaded);
 
         return true;
     }
 
+    /**
+     * Used to set the viability of a menu item.
+     * This method also does some null-checking so that neither menu nor the menu item are null
+     * in order to prevent nullpointer exceptions.
+     * @param menu The menu that should be used
+     * @param  menuId The id of the menu item that will be used
+     * @param visibility The new visibility status of given menu item
+     * */
+    private static void setItemVisibility(Menu menu, int menuId, boolean visibility) {
+        if(menu == null) {
```

Comment:
Please add space

Reply:
Sorry, I have more complaints :/

This is still missing a space. Also, please add braces to the if statement in the method below.

Reply:
No need to be sorry.
Should be all done now. I also added spaces to a couple if statements missing them.